spectral_cube/spectral_cube.py

```diff
@@ -2363,6 +2363,8 @@ def __init__(self, data, wcs, mask=None, meta=None, fill_value=np.nan,
             self.pixels_per_beam = (self.beam.sr /
                                     (astropy.wcs.utils.proj_plane_pixel_area(self.wcs) *
                                      u.deg**2)).to(u.dimensionless_unscaled).value
+        self.abort = False  # Checked mid calculation
```

Comment:
It looks like `abort` should be a local variable, not a class instance attribute.  Is there any reason to define this as `self.abort` instead of just `abort`?

Reply:
I see the `abort_function` below; is the idea that this function would be called and somehow update the state of the `Cube` object *during* the smoothing operation?  I was unaware that that is possible.

Reply:
Thank you for the quick review. Yes this is exactly it does. During the smoothing operation, a separate thread with an abort button can call `abort_function` which in turn would toggle the value of `self.abort` to true. If we make these variables local to the function, we would not be able to externally change their states.